tests/integration/conftest.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm integration test config."""
+
+import asyncio
+import logging
+
+import pytest_asyncio
+from helpers import (
+    APP_NAME_AIRBYTE_SERVER,
+    APP_NAME_TEMPORAL_ADMIN,
+    APP_NAME_TEMPORAL_SERVER,
+    create_default_namespace,
+    get_airbyte_charm_resources,
+    perform_airbyte_integrations,
+    perform_temporal_integrations,
+)
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+
+@pytest_asyncio.fixture(name="deploy", scope="module")
+async def deploy(ops_test: OpsTest):
+    """Test the app is up and running."""
+    charm = await ops_test.build_charm(".")
+    resources = get_airbyte_charm_resources()
+
+    asyncio.gather(
+        ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME_AIRBYTE_SERVER, trust=True),
+        ops_test.model.deploy(
+            APP_NAME_TEMPORAL_SERVER,
+            channel="edge",
+            config={"num-history-shards": 1},
+        ),
+        ops_test.model.deploy(APP_NAME_TEMPORAL_ADMIN, channel="edge"),
+        ops_test.model.deploy("postgresql-k8s", channel="14/stable", trust=True),
+        ops_test.model.deploy("minio", channel="edge"),
+    )
+
+    async with ops_test.fast_forward():
```

Comment:
Just a heads up, I've had issues wrapping "wait_for_idle" with fast_forward. Fast_forward worked well when I used it to wrap the deploy step but what happened with wait_for_idle was specifically with the traefik-k8s charm, it was updating the status every update interval (something like every 5-10s) so it was never "idle" which checks if the charm status hasn't changed in ~15s.